src/water_cmd.cpp

```diff
@@ -452,10 +454,18 @@ CommandCost CmdBuildCanal(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32
 	TileArea ta(tile, p1);
 
 	/* Outside the editor you can only drag canals, and not areas */
-	if (_game_mode != GM_EDITOR && ta.w != 1 && ta.h != 1) return CMD_ERROR;
+	if (_game_mode != GM_EDITOR) {
+		TileArea ta(tile, p1);
+		if (ta.w != 1 && ta.h != 1) return CMD_ERROR;
+	}
 
 	CommandCost cost(EXPENSES_CONSTRUCTION);
-	TILE_AREA_LOOP(tile, ta) {
+
+	std::unique_ptr<TileIterator> iter;
+	if (HasBit(p2, 2)) iter = std::make_unique<DiagonalTileIterator>(tile, p1); else iter = std::make_unique<OrthogonalTileIterator>(tile, p1);
```

Comment:
Nope. Don't try to circumvent the codestyle, use braces and indentation properly

Reply:
D'oh. I copied from IRC and assumed this was like the `if (condition) return false;` single-line style used elsewhere.